### PR TITLE
[KUBOS-350] Linux telemetry service

### PR DESCRIPTION
**Pull request by plauche**
* Executable telemetry service for the linux environment
* Most basic level of functionality, allows for subscribers and publishers, but does not come with any built-in

### CODE REVIEW

File: linux-telemetry-service/source/main.c [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+#include <telemetry-linux/server.h>
+#include <telemetry/telemetry.h>
+
+int main(int argc, char ** argv)
+{
+    socket_conn server_conn;
+    kprv_socket_server_setup(&server_conn, TELEMETRY_SOCKET_PORT, TELEMETRY_SUBSCRIBERS_MAX_NUM);
+    bool running = true;
+
+    while (running)
```
Comment (CatFreed) [inline linux-telemetry-service/source/main.c]: How does running get changed? Would setting up a catch for SIGTERM be appropriate?

File: linux-telemetry-service/source/main.c [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    {
+        socket_conn conn;
+        while (!kprv_socket_server_accept(&server_conn, &conn))
+        {
+            continue;
```
Comment (CatFreed) [inline linux-telemetry-service/source/main.c]: A sleep might be good, unless there's no risk of the accept taking up too much system time

Comment (plauche) [inline linux-telemetry-service/source/main.c]: The accept is currently a blocking call, so I don't think adding a sleep here will make much difference.

File: linux-telemetry-service/source/main.c (hunk 1 above)
Comment (plauche) [inline linux-telemetry-service/source/main.c]: I'll add the basic bones (non-functioning) for a more graceful shutdown using the signal catcher, but currently we block on the socket, so we can't really gracefully bring down the process. We should look into reworking the base tcp interface to use a non-blocking connection.

Comment (CatFreed) [inline linux-telemetry-service/source/main.c]: :+1:

File: linux-telemetry-service/source/main.c (hunk 2 above)
Comment (CatFreed) [inline linux-telemetry-service/source/main.c]: :+1:

**Pull request merged**